README.md

```diff
@@ -88,7 +89,22 @@ The `timeout` and `retry` values, as show in in the `pip` service section, are o
 
 ### Custom sources and layers
 
-Pelias allows importing your own data with custom values for `source` and `layer`, however you MUST tell Pelias about them via `pelias.json` using the `targets.layers_by_source`, `targets.source_aliases` and `targets.layer_aliases` configuration parameters.
+Pelias allows importing your own data with custom values for `source` and `layer`.
+
+Custom sources and layers are not automatically detected, you MUST set `targets.auto_discover` to `true` in your `pelias.json` to make Pelias aware of them.
+
+The `auto_discover` functionality sends a request to elasticsearch in order to automatically discover sources and layers from elasticsearch when the API server starts-up.
+
+Be aware that the query sent to elasticsearch can take several seconds to execute the first time against a large index, the query is cached in elasticsearch for subsequent requests.
+
+If you are importing custom layers and are running a city or small region sized build then the start-up delay will likely not affect you, you can safely use `targets.auto_discover:true`.
```

Comment:
This reads like the problem with the query to Elasticsearch is that it causes the API to wait around for a while. Really the danger is that it puts a lot of load on the Elasticsearch cluster. So maybe this should read more like

>Be aware that the query sent to Elasticsearch can take several seconds to execute the first time against a large index, potentially impacting the impact on other queries hitting Elasticsearch. The query is cached in Elasticsearch for subsequent requests.
> 
> If you are importing custom layers and are running a city or small region sized build then the impact of this query will likely be too small to notice, you can safely use `targets.auto_discover:true`